content/blog/thanos-monitoring.en.md

```diff
@@ -0,0 +1,524 @@
+"pushing metrics" part in this article, you can read about the pros and cons of
+pushing metrics
+[here](https://docs.google.com/document/d/1H47v7WfyKkSLMrR8_iku6u9VB73WrVzBHb2SB6dL9_g/edit#heading=h.2v27snv0lsur).
+It is recommended to push metrics as a last resort or when not trusting multiple
+cluster or tenant (for example when building a prometheus as a service
+offering). Anyway this might be a topic for a further article but we will focus
+on the scrapping here.
+
+### Enter Thanos
+
+![](/images/logos/thanos.svg#logosize)
+
+[Thanos](https://thanos.io/) is a "*Open source, highly available Prometheus
+setup with long term storage capabilities*". Thanos is used by a lot of well
+known companies. It is also a part of [the CNCF incubating
+projects](https://www.cncf.io/projects/).
+
+One of the main feature of Thanos is to allow for "unlimited" storage. How so ?
+By using object storage (such as S3), which is offered by almost every cloud
+provider. If running on premises, object storage can be offered with solution
+like [rook](https://rook.io/) or [minio](https://min.io/).
+
+#### How does it work ?
+
+Thanos is running alongside prometheus. It is common to start with a Prometheus
+only setup and to upgrade to a Thanos one.
+
+Thanos is split into several components, each having one goal (as every service
+should be :) ). The component communicate with each other through gRPC.
+
+##### Thanos Sidecar
+
+![](/images/prometheus-thanos/thanos-sidecar-no-background.png)
+
+Thanos is running alongside prometheus (with a sidecar) and export prometheus
+metrics every 2h to an object storage. This allow prometheus to be *almost*
+stateless. Prometheus is still keeping 2 hours worth of metrics in memory so you
+might still loose 2 hours worth of metrics in case of outage (this is problem
+which should be handle by your prometheus setup, with HA/Sharding, and not by
+Thanos).
+
+[Thanos sidecar](https://thanos.io/tip/components/sidecar.md/) is available out
+of the box with [Prometheus
+Operator](https://github.com/prometheus-operator/prometheus-operator) and [Kube
+Prometheus
+Stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack)
+and can be deploy easily. This component act as a store for Thanos Query.
+
+##### Thanos Store
+
+[Thanos store](https://thanos.io/tip/components/store.md/) acts a gateway to
+translate query to remote object storage. It can also cache some information on
+local storage. Basically this is the component that allows you to query an
+object store for metrics. This component acts as a store for Thanos Query
+
+##### Thanos Compactor
+
+[Thanos compactor](https://thanos.io/tip/components/compact.md/) is a singleton
+(it is not scalable) that is responsible for compacting and downsampling the
+metrics stored inside an object store. Downsampling is the fact of loosing
+granularity on your metrics over time: for example you may want to keep your
```

Comment:
`time: for example you may`

maybe one of:
- `time. You may`
- `time. For example you may`